sale_order_recurrence/models/sale_order_recurrence_wizard.py

```diff
@@ -0,0 +1,126 @@
+# Copyright (C) 2015 - Today: GRAP (http://www.grap.coop)
+# @author: Sylvain LE GAL (https://twitter.com/legalsylvain)
+# @author: Quentin DUPONT (https://twitter.com/pondupont)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+import datetime
+
+from dateutil.relativedelta import relativedelta
+
+from odoo import api, fields, models
+
+
+class SaleOrderRecurrenceWizard(models.TransientModel):
+    _name = "sale.order.recurrence.wizard"
+    _description = "Sale Order Duplication Wizard"
+
+    _RECURRENCE_TYPE_KEYS = [
+        ("week", "Weekly"),
+        ("month", "Monthly"),
+    ]
+
+    # Column Section
+    order_id = fields.Many2one(
+        comodel_name="sale.order",
+        string="Sale Order",
+        readonly=True,
+    )
+
+    partner_id = fields.Many2one(
+        comodel_name="res.partner",
+        string="Partner",
+        readonly=True,
+    )
+
+    begin_date = fields.Date(
+        required=True,
+    )
+
+    include_current_date = fields.Boolean(default=False)
+
+    recurrence_type = fields.Selection(
+        selection=_RECURRENCE_TYPE_KEYS,
+        default="week",
+        required=True,
+    )
+
+    recurrence_duration = fields.Integer(required=True, default=0)
+
+    date_line_ids = fields.One2many(
+        comodel_name="sale.order.recurrence.wizard.date.line",
+        inverse_name="wizard_id",
+        string="New Dates",
+    )
+
+    @api.model
+    def default_get(self, fields_list):
+        res = super().default_get(fields_list)
+        order_id = self.env.context.get("active_id", False)
+        if not order_id:
+            res.update({"order_id": False})
```

Comment:
FYI, this is more efficient:

```suggestion
            res.update({
                "order_id": False,
                "partner_id": False,
                "begin_date": False,
            })
```